examples/vector_return/src/main.c

```diff
@@ -0,0 +1,22 @@
+#include <stdio.h>
+#include <stdint.h>
+
+extern size_t
+counter_generate(size_t start, size_t size, size_t **vec);
+
+extern void
+counter_free(size_t *vec, size_t size);
+
+int main(void) {
+  size_t *vec;
+
+  counter_generate(0, 10, &vec);
```

Comment:
The return value is being returned. This needs to match up with the future call to `free`.